crates/bitwarden-error-macro/src/flat/attribute.rs

```diff
@@ -0,0 +1,117 @@
+use quote::quote;
+use syn::Data;
+
+pub(crate) fn bitwarden_error_flat(
+    input: &syn::DeriveInput,
+    type_identifier: &proc_macro2::Ident,
+    export_as_identifier: &proc_macro2::Ident,
+) -> proc_macro::TokenStream {
+    match &input.data {
+        Data::Enum(data) => {
+            let variant_names = data.variants.iter().map(|variant| &variant.ident);
+
+            let match_arms = data.variants.iter().map(|variant| match variant.fields {
+                syn::Fields::Unit => {
+                    let variant_ident = &variant.ident;
+                    let variant_name = format!("{}", variant_ident);
```

Comment:
There's a bit of repetition with these lines for all the branches, can they be moved outside the match?

Also small subjective choice, but `variant_ident.to_string()` is the same as `format!("{}", variant_ident)`, but I think it looks nicer

Reply:
Good catch, I'll fix both!